core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1046,7 +1049,9 @@ class ReplicaManager(val config: KafkaConfig,
     //                        3) has enough data to respond
     //                        4) some error happens while reading data
     //                        5) we found a diverging epoch
-    if (timeout <= 0 || fetchInfos.isEmpty || bytesReadable >= fetchMinBytes || errorReadingData || hasDivergingEpoch) {
+    //                        6) the preferred read replica not local replica
```

Comment:
nit: `has a preferred read replica`?

Reply:
done

Reply:
Following your suggestion, I modified the unit test
I also found some problems when modifying the unit test, which I think are bugs